src/main/java/uk/sky/cqlmigrate/SchemaUpdates.java

```diff
@@ -77,22 +76,23 @@ void add(String filename, Path path) {
 
     private String calculateChecksum(Path path) {
         try {
-            return new PathByteSource(path).hash(Hashing.sha1()).toString();
-        } catch (IOException e) {
-            throw Throwables.propagate(e);
+            final MessageDigest digest = MessageDigest.getInstance("SHA-1");
+            final byte[] hash = digest.digest(Files.readAllBytes(path));
+            return bytesToHex(hash);
+        } catch (IOException | NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
         }
     }
 
-    private static class PathByteSource extends ByteSource {
-        private final Path path;
-
-        public PathByteSource(Path path) {
-            this.path = path;
-        }
-
-        @Override
-        public InputStream openStream() throws IOException {
-            return java.nio.file.Files.newInputStream(path);
+    // see https://stackoverflow.com/a/9855338/627727 for where this comes from
+    private final static char[] hexArray = "0123456789abcdef".toCharArray();
+    private static String bytesToHex(byte[] bytes) {
+        char[] hexChars = new char[bytes.length * 2];
+        for (int j = 0; j < bytes.length; j++) {
+            int v = bytes[j] & 0xFF;
```

Comment:
I'm not sure if you need this intermediate integer, or the `hexArray` array.

```java
        StringBuilder buf = new StringBuilder(2 * bytes.length);
        for (byte b : bytes) {
            buf.append(Character.forDigit(hexArray[b >>> 4], 16))
               .append(Character.forDigit(hexArray[b & 0x0F], 16));
        }
        return buf.toString();
```

As all of the hex digits are in Latin-1, the use of `StringBuilder` is equivalent to having an intermediate array (eg. `hexChars`).

Reply:
That seems right, I'll try to change it.

I pretty much just took the Stack Overflow answer because I didn't want to think about it :laughing:

Reply:
I've updated this to follow the sort of style you were suggesting. The intermediate value (I believe) is merely to get the byte as unsigned before getting the 4 first bits, and the last 4 bits.

I added an intermediate `final int asUnsigned = Byte.toUnsignedInt(b);` before the code you suggested above